src/Features/Lsif/Generator/Generator.cs

```diff
@@ -195,6 +195,21 @@ public void GenerateForCompilation(Compilation compilation, string projectPath,
                         var referenceResultsId = symbolResultsTracker.GetResultIdForSymbol(referencedSymbol.OriginalDefinition, Methods.TextDocumentReferencesName, () => new ReferenceResult(idFactory));
                         lsifJsonWriter.Write(new Item(referenceResultsId.As<ReferenceResult, Vertex>(), lazyRangeVertex.Value.GetId(), documentVertex.GetId(), idFactory, property: "references"));
                     }
+
+                    // Write hover information for the symbol, if edge has not already been added.
+                    // 'textDocument/hover' edge goes from the symbol ResultSet vertex to the hover result
+                    // See https://github.com/Microsoft/language-server-protocol/blob/main/indexFormat/specification.md#resultset for an example.
+                    if (symbolResultsTracker.ResultSetNeedsInformationalEdgeAdded(symbolForLinkedResultSet, Methods.TextDocumentHoverName))
+                    {
+                        // TODO: Can we avoid the WaitAndGetResult_CanCallOnBackground call by adding a sync method to compute hover?
```

Comment:
or just make the lsif generator async.

Reply:
I had prototyped that earlier, but Jason recommended against it as we want to eventually do execute LSIF generation from an analyzer.